documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -0,0 +1,80 @@
+[id='file-stores_{context}']
+= File-Based Cache Stores
+
+File-based cache stores provide persistent storage on the local host filesystem where {brandname} is running.
+For clustered caches, file-based cache stores are unique to each {brandname} node.
+
+[WARNING]
+====
+Never use filesystem-based cache stores on shared file systems, such as an NFS or Samba share, because they do not provide file locking capabilities and data corruption can occur.
+
+Additionally if you attempt to use transactional caches with shared file systems, unrecoverable failures can happen when writing to files during the commit phase.
+====
+
+[discrete]
+== Soft-Index File Stores
+
+`SoftIndexFileStore` is the default implementation for file-based cache stores and stores data in a set of append-only files.
+
+When append-only files:
+
+* Reach their maximum size, {brandname} creates a new file and starts writing to it.
+* Reach the compaction threshold of less than 50% usage, {brandname} overwrites the entries to a new file and then deletes the old file.
+
+.B+ trees
+
+To improve performance, append-only files in a `SoftIndexFileStore` are indexed using a **B+ Tree** that can be stored both on disk and in memory.
+The in-memory index uses Java soft references to ensure it can be rebuilt if removed by Garbage Collection (GC) then requested again.
+
+Because `SoftIndexFileStore` uses Java soft references to keep indexes in memory, it helps prevent out-of-memory exceptions.
+GC removes indexes before they consume too much memory while still falling back to disk.
+
+You can configure any number of B+ trees.
+By default {brandname} creates up to 16 B+ trees, which means there can be up to 16 indexes.
+Having multiple indexes prevents bottlenecks from concurrent writes to an index.
+It also minimizes the number of entries that can be read at the same time because {brandname} reads all entries at the same time when it iterates over a soft-index file store.
+Fewer indexes means that {brandname} must perform more reads.
+
+Each entry in the B+ tree is a node.
+By default, the size of each node is limited to 4096 bytes.
+This size also limits the key length, more precisely the length of serialized keys, which means you cannot use keys longer than the size of the node, 15
+bytes.
+Additionally, key length is stored as "short", which limits key length to 32767 bytes.
```

Comment:
The key cannot be larger than the node size less than the some reserved space, so in that case the key by default can't be larger than 4096 or so.